{
	"document_title": "Security update for python",
	"document_type": "SUSE Patch",
	"document_publisher": {
		"type": "Vendor",
		"contact_details": "security@suse.de",
		"issuing_authority": "SUSE Security Team"
	},
	"document_tracking": {
		"identification": {
			"id": "SUSE-SU-2023:0213-1"
		},
		"status": "Final",
		"version": "1",
		"revision_history": {
			"revision": {
				"number": "1",
				"date": "2023-01-30T16:27:03Z",
				"description": "current"
			}
		},
		"initial_release_date": "2023-01-30T16:27:03Z",
		"current_release_date": "2023-01-30T16:27:03Z",
		"generator": {
			"engine": "cve-database/bin/generate-cvrf.pl",
			"date": "2017-02-24T01:00:00Z"
		}
	},
	"document_notes": [
		{
			"text": "Security update for python",
			"title": "Topic",
			"type": "Summary"
		},
		{
			"text": "This update for python fixes the following issues:\n\n- CVE-2022-45061: Fixed an excessive CPU usage when decoding crafted\n  IDNA domain names (bsc#1205244).\n\nNon-security fixes:\n\n- Fixed the 2038 bug in the compileall module (bsc#1202666).\n",
			"title": "Details",
			"type": "General"
		},
		{
			"text": "The CVRF data is provided by SUSE under the Creative Commons License 4.0 with Attribution (CC-BY-4.0).",
			"title": "Terms of Use",
			"type": "Legal Disclaimer"
		},
		{
			"text": "SUSE-2023-213,SUSE-SLE-SERVER-12-SP2-BCL-2023-213",
			"title": "Patchnames",
			"type": "Details"
		}
	],
	"documentdistribution": "Copyright SUSE LLC under the Creative Commons License 4.0 with Attribution (CC-BY-4.0)",
	"document_references": [
		{
			"type": "Self",
			"url": "https://www.suse.com/support/update/announcement/2023/suse-su-20230213-1/",
			"description": "Link for SUSE-SU-2023:0213-1"
		},
		{
			"type": "Self",
			"url": "https://lists.suse.com/pipermail/sle-security-updates/2023-January/013594.html",
			"description": "E-Mail link for SUSE-SU-2023:0213-1"
		},
		{
			"type": "Self",
			"url": "https://www.suse.com/support/security/rating/",
			"description": "SUSE Security Ratings"
		},
		{
			"type": "Self",
			"url": "https://bugzilla.suse.com/1202666",
			"description": "SUSE Bug 1202666"
		},
		{
			"type": "Self",
			"url": "https://bugzilla.suse.com/1205244",
			"description": "SUSE Bug 1205244"
		},
		{
			"type": "Self",
			"url": "https://www.suse.com/security/cve/CVE-2022-45061/",
			"description": "SUSE CVE CVE-2022-45061 page"
		}
	],
	"product_tree": {
		"branch": [
			{
				"type": "Product Family",
				"name": "SUSE Linux Enterprise Server 12 SP2-BCL",
				"branch": [
					{
						"type": "Product Name",
						"name": "SUSE Linux Enterprise Server 12 SP2-BCL",
						"full_product_name": {
							"text": "SUSE Linux Enterprise Server 12 SP2-BCL",
							"product_id": "SUSE Linux Enterprise Server 12 SP2-BCL",
							"cpe": "cpe:/o:suse:sles-bcl:12:sp2"
						}
					}
				],
				"full_product_name": {}
			},
			{
				"type": "Product Version",
				"name": "libpython2_7-1_0-2.7.18-28.93.1",
				"full_product_name": {
					"text": "libpython2_7-1_0-2.7.18-28.93.1",
					"product_id": "libpython2_7-1_0-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "libpython2_7-1_0-32bit-2.7.18-28.93.1",
				"full_product_name": {
					"text": "libpython2_7-1_0-32bit-2.7.18-28.93.1",
					"product_id": "libpython2_7-1_0-32bit-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "libpython2_7-1_0-64bit-2.7.18-28.93.1",
				"full_product_name": {
					"text": "libpython2_7-1_0-64bit-2.7.18-28.93.1",
					"product_id": "libpython2_7-1_0-64bit-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-2.7.18-28.93.1",
					"product_id": "python-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-32bit-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-32bit-2.7.18-28.93.1",
					"product_id": "python-32bit-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-64bit-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-64bit-2.7.18-28.93.1",
					"product_id": "python-64bit-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-base-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-base-2.7.18-28.93.1",
					"product_id": "python-base-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-base-32bit-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-base-32bit-2.7.18-28.93.1",
					"product_id": "python-base-32bit-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-base-64bit-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-base-64bit-2.7.18-28.93.1",
					"product_id": "python-base-64bit-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-curses-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-curses-2.7.18-28.93.1",
					"product_id": "python-curses-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-demo-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-demo-2.7.18-28.93.1",
					"product_id": "python-demo-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-devel-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-devel-2.7.18-28.93.1",
					"product_id": "python-devel-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-doc-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-doc-2.7.18-28.93.1",
					"product_id": "python-doc-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-doc-pdf-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-doc-pdf-2.7.18-28.93.1",
					"product_id": "python-doc-pdf-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-gdbm-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-gdbm-2.7.18-28.93.1",
					"product_id": "python-gdbm-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-idle-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-idle-2.7.18-28.93.1",
					"product_id": "python-idle-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-strict-tls-check-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-strict-tls-check-2.7.18-28.93.1",
					"product_id": "python-strict-tls-check-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-tk-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-tk-2.7.18-28.93.1",
					"product_id": "python-tk-2.7.18-28.93.1"
				}
			},
			{
				"type": "Product Version",
				"name": "python-xml-2.7.18-28.93.1",
				"full_product_name": {
					"text": "python-xml-2.7.18-28.93.1",
					"product_id": "python-xml-2.7.18-28.93.1"
				}
			}
		],
		"relationship": [
			{
				"product_reference": "libpython2_7-1_0-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "libpython2_7-1_0-32bit-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-32bit-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-base-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-base-32bit-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-curses-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-demo-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-doc-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-doc-pdf-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-gdbm-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-idle-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-tk-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			},
			{
				"product_reference": "python-xml-2.7.18-28.93.1",
				"relation_type": "Default Component Of",
				"relates_to_product_reference": "SUSE Linux Enterprise Server 12 SP2-BCL"
			}
		]
	},
	"vulnerability": [
		{
			"notes": [
				{
					"text": "An issue was discovered in Python before 3.11.1. An unnecessary quadratic algorithm exists in one path when processing some inputs to the IDNA (RFC 3490) decoder, such that a crafted, unreasonably long name being presented to the decoder could lead to a CPU denial of service. Hostnames are often supplied by remote servers that could be controlled by a malicious actor; in such a scenario, they could trigger excessive CPU consumption on the client attempting to make use of an attacker-supplied supposed hostname. For example, the attack payload could be placed in the Location header of an HTTP response with status code 302. A fix is planned in 3.11.1, 3.10.9, 3.9.16, 3.8.16, and 3.7.16.",
					"title": "Vulnerability Description",
					"type": "General"
				}
			],
			"cve": "CVE-2022-45061",
			"product_statuses": [
				{
					"type": "Fixed",
					"product_id": [
						"SUSE Linux Enterprise Server 12 SP2-BCL:libpython2_7-1_0-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:libpython2_7-1_0-32bit-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-32bit-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-base-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-base-32bit-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-curses-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-demo-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-doc-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-doc-pdf-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-gdbm-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-idle-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-tk-2.7.18-28.93.1",
						"SUSE Linux Enterprise Server 12 SP2-BCL:python-xml-2.7.18-28.93.1"
					]
				}
			],
			"threats": [
				{
					"type": "Impact",
					"description": "moderate"
				}
			],
			"cvss_score_sets": {
				"score_set_v_2": {},
				"score_set_v_3": {
					"base_score_v_3": "6.5",
					"vector_v_3": "CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:N/I:N/A:H"
				},
				"scoreset": {}
			},
			"remediations": [
				{
					"type": "Vendor Fix",
					"description": {
						"text": "To install this SUSE Security Update use the SUSE recommended installation methods like YaST online_update or \"zypper patch\".\n",
						"lang": "en"
					},
					"url": "https://www.suse.com/support/update/announcement/2023/suse-su-20230213-1/"
				}
			],
			"references": [
				{
					"url": "https://www.suse.com/security/cve/CVE-2022-45061.html",
					"description": "CVE-2022-45061"
				},
				{
					"url": "https://bugzilla.suse.com/1205244",
					"description": "SUSE Bug 1205244"
				},
				{
					"url": "https://bugzilla.suse.com/1211488",
					"description": "SUSE Bug 1211488"
				}
			]
		}
	]
}
